src/Hl7.Fhir.Specification/Specification/Navigation/StructureDefinitionWalker.cs

```diff
@@ -141,9 +163,9 @@ public IEnumerable<StructureDefinitionWalker> Expand()
             else if (Current.Current.Type.Count >= 1)
             {
                 return Current.Current.Type
-                    .Select(t => t.GetTypeProfile())
+                    .Select(t => (profile: t.GetTypeProfile(), targetProfiles: t.TargetProfile))
```

Comment:
Multiple targetProfiles will always cause an ambiguous discriminator, so I think we should only allow one single target profile?

Reply:
And throw an exception when there are more than 1 target profiles? In that case we can also make the private field `_targetProfiles` a string instead of an array of strings.

Reply:
Hmm, that is not possible, because this code is part of the method `Expand()` which is used elsewhere as well.

Reply:
Looking at the code for the second time, I think the initial code is correct. Allowing multiple targetProfiles will result in multiple `StructureDefinitionWalker`'s in function `Resolve()`, which is still allowed, because the profile author can decide to concatenate other functions (like `first()`) on to these walkers.

In the method `DiscriminatorFactory.walkToCondition` is the final check whether there are more than 1 conditions. In that case an exception will be thrown.